seoyoung059/Week_06/BOJ_14865/BOJ_14865.java

```diff
@@ -0,0 +1,94 @@
+package Week_06.BOJ_14865;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.ArrayDeque;
+import java.util.PriorityQueue;
+import java.util.Stack;
+import java.util.StringTokenizer;
+
+public class BOJ_14865 {
+
+
+    static class P implements Comparable {
+        int s;
+        int e;
+
+        P(int s, int e) {
+            this.s = s;
+            this.e = e;
+        }
+
+        @Override
+        public int compareTo(Object o) {
+            return this.s - ((P) o).s;
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        int n = Integer.parseInt(br.readLine());
+        StringTokenizer st;
+
+        int tmp = Integer.MIN_VALUE;
+        int prevy = 0;
+        PriorityQueue<P> pq = new PriorityQueue<>();
+
+        int y, x = 0;
+        int sy = 0, sx = 0;
+        boolean find = false;
+        int firstX = Integer.MIN_VALUE;
+        boolean first = true;
+        for (int i = 0; i < n + 1; i++) {
+            if (i < n) {
+                st = new StringTokenizer(br.readLine());
+                x = Integer.parseInt(st.nextToken());
+                y = Integer.parseInt(st.nextToken());
+                if (first) {
+                    sy = y;
+                    sx = x;
+                    first = false;
+                }
+            } else {
+                y = sy;
+                x = sx;
+            }
+            if (prevy < 0 && y > 0) {
+                tmp = x;
+                find = true;
+            } else if (prevy > 0 && y < 0) {
+                if (find) {
+                    pq.offer((tmp < x) ? new P(tmp, x) : new P(x, tmp));
+                    find = false;
+                } else {
+                    firstX = x;
+                }
+            }
+            prevy = y;
+        }
```

Comment:
혹시나 메모리나 시간을 낮추고 좀 더 가독성을 챙기고 싶으시다면 아래와 같은 코드도 추천드립니다!
사람마다 가독성이 좋다고 느끼는 코드는 다를 수도 있으니깐, 이런 방식도 있구나라고 참고만 해주시면 좋을 것 같아요😃

```suggestion
        int[] x = new int[n], y = new int[n];
        boolean find = false, first = true;
        int firstX = Integer.MIN_VALUE;
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            x[i] = Integer.parseInt(st.nextToken());
            y[i] = Integer.parseInt(st.nextToken());
        }
        for (int i = 0; i < n + 1; i++) {
            int curr_x = x[(i + 1) % n], curr_y = y[(i + 1) % n];
            if (prev_y < 0 && curr_y > 0) {
                tmp = x;
                find = true;
            } else if (prev_y > 0 && curr_y < 0) {
                if (find) {
                    pq.offer((tmp < curr_x) ? new P(tmp, curr_x) : new P(curr_x, tmp));
                    find = false;
                } else {
                    first_x = curr_x;
                }
            }
            prev_y = curr_y;
        }
```